### ISSUE TITLE
v4.0.0-alpha.2  pull-right  not working

### ISSUE THREAD

**Issue by iloveyo123u1** [twbs/bootstrap#19951]
see  http://plnkr.co/edit/lYHw8cd4uXq0VL97XoqU?p=preview

**Comment by twbs-lmvtfy** [comment]
Hi @iloveyo123u1!

You appear to have posted a live example (http://run.plnkr.co/plunks/lYHw8cd4uXq0VL97XoqU/), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* [W001](https://github.com/twbs/bootlint/wiki/W001): `<head>` is missing UTF-8 charset `<meta>` tag
* [W002](https://github.com/twbs/bootlint/wiki/W002): `<head>` is missing X-UA-Compatible `<meta>` tag that disables old IE compatibility modes
* [W003](https://github.com/twbs/bootlint/wiki/W003): `<head>` is missing viewport `<meta>` tag that enables responsiveness

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by iloveyo123u1** [comment]
```
<!DOCTYPE html>
<html>
<head>
  <link rel="stylesheet" href="http://v4-alpha.getbootstrap.com/dist/css/bootstrap.min.css" />
</head>
<body>
  <div class="container">
    <ul class="list-group">
      <li class="list-group-item">
        <h2>info</h2></li>
      <li class="list-group-item"><span class="label label-default label-pill pull-right">2012-11-12</span>the date pull-right not working</li>
      <li class="list-group-item">11111 </li>
      <li class="list-group-item">11111</li>
      <li class="list-group-item">11111</li>
      <li class="list-group-item">11111</li>
      <li class="list-group-item">11111</li>
      <li class="list-group-item">11111</li>
    </ul>
  </div>
  <!-- /.container -->
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.4/jquery.min.js"></script>
  <script src="http://v4-alpha.getbootstrap.com/dist/js/bootstrap.min.js"></script>
</body>
</html>
```

**Comment by mdo** [comment]
Use `.pull-xs-right`—the class was changed. Might try to restore those non-tier classes, but that's another issue altogether.

**Comment by LeftPinkie** [comment]
I am having strange alignment issues when using `.pull-xs-right` with `.list-group-item` and `.p-a-0`.

- The `.pull-xs-right` text does not appear if the `<li>` is wrapped to new line.
- The  `.pull-xs-right` does not float all the way to right when I resize my window... there is extra spacing between the `.pull-xs-right` text and the parent right margin.

**Comment by gbrigens** [comment]
Am having the same issue here is sample of my code just as from bootstrap 4 documentation.`<nav class="navbar navbar-dark bg-primary">
    <button class="navbar-toggler hidden-sm-up" type="button" data-toggle="collapse" data-target="#exCollapsingNavbar2">
    &#9776;
    </button>
    <div class="collapse navbar-toggleable-xs" id="exCollapsingNavbar2">
        <a class="navbar-brand" [routerLink]="['/']">Recipe Book</a>
        <ul class="nav navbar-nav">
            <li class="nav-item" routerLinkActive="active">
                <a [routerLink]="['/recipe']" class="nav-link">Recipe</a>
            </li>
            <li class="nav-item" routerLinkActive="active">
                <a [routerLink]="['/shopping-list']" class="nav-link">Shopping List</a>
            </li>
            <li class="nav-item btn-group pull-xs-right">
                <a class="dropdown-toggle nav-link" type="button" id="dropdownMenu1" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
                User
                </a>
                <div class="dropdown-menu" aria-labelledby="dropdownMenu1">
                    <a class="dropdown-item" href="#">Login</a>
                    <a class="dropdown-item" href="#">Other action</a>
                    <a class="dropdown-item" href="#">Something else here</a>
                    <a class="dropdown-item" href="#">Log Out</a>
                </div>
            </li>

        </ul>
    </div>
</nav>`
